packages/editor/src/components/autosave-monitor/index.js

```diff
@@ -42,12 +42,25 @@ export class AutosaveMonitor extends Component {
 	}
 
 	toggleTimer( isPendingSave ) {
-		clearTimeout( this.pendingSave );
-		const { autosaveInterval } = this.props;
-		if ( isPendingSave ) {
+		const { interval, shouldThrottle = false } = this.props;
+
+		// By default, AutosaveMonitor will wait for a pause in editing before
+		// autosaving. In other words, its action is "debounced".
+		//
+		// The `shouldThrottle` props allows overriding this behaviour, thus
+		// making the autosave action "throttled".
+		if ( ! shouldThrottle ) {
+			clearTimeout( this.pendingSave );
```

Comment:
Should `clearTimeout` call be guarded here to avoid executing this block when `this.pendingSave` is not defined?